components/camel-spark/src/test/java/org/apache/camel/component/spark/SparkProducerTest.java

```diff
@@ -239,27 +245,18 @@ public void shouldExecuteHiveCountQuery() {
     @Test
     public void shouldCountFrame() {
         assumeTrue(shouldRunHive);
-        DataFrameCallback callback = new DataFrameCallback<Long>() {
```

Comment:
Why you have removed async callback test after the correctly changed DataFrameCallback?
It's better to have both sync and async tests.